### PR TITLE
Fix retriever

### PR DESCRIPTION
**Pull request by jerpint**

### CODE REVIEW

File: buster/retriever/base.py [hunk 1]
```diff
@@ -36,27 +40,39 @@ def get_embedding(query: str, engine: str):
...
 
-    def retrieve(self, query: str, source: str = None) -> pd.DataFrame:
-        top_k = self.top_k
-        thresh = self.thresh
-        query_embedding = self.get_embedding(query, engine=self.embedding_model)
+    def get_topk_documents(self, query: str, source: str = None, top_k: int = None) -> pd.DataFrame:
+        """Get the topk documents matching a user's query.'
 
-        documents = self.get_documents(source)
+        If no mathces are found, returns an empty dataframe."""
```
Comment (hbertrand) [inline buster/retriever/base.py:46]: typo

File: buster/retriever/base.py [hunk 2]
```diff
@@ -36,27 +40,39 @@ def get_embedding(query: str, engine: str):
         logger.info("generating embedding")
         return get_embedding(query, engine=engine)
 
-    def retrieve(self, query: str, source: str = None) -> pd.DataFrame:
-        top_k = self.top_k
-        thresh = self.thresh
-        query_embedding = self.get_embedding(query, engine=self.embedding_model)
+    def get_topk_documents(self, query: str, source: str = None, top_k: int = None) -> pd.DataFrame:
+        """Get the topk documents matching a user's query.'
```
Comment (hbertrand) [inline buster/retriever/base.py:44]: leftover ' at the end

File: buster/retriever/base.py [hunk 3]
```diff
@@ -36,27 +40,39 @@ def get_embedding(query: str, engine: str):
         logger.info("generating embedding")
         return get_embedding(query, engine=engine)
 
-    def retrieve(self, query: str, source: str = None) -> pd.DataFrame:
-        top_k = self.top_k
-        thresh = self.thresh
-        query_embedding = self.get_embedding(query, engine=self.embedding_model)
+    def get_topk_documents(self, query: str, source: str = None, top_k: int = None) -> pd.DataFrame:
```
Comment (hbertrand) [inline buster/retriever/base.py:43]: should be `@abstractmethod`

**Review by hbertrand — CHANGES_REQUESTED** [review]
It's weird to have `topk` in both the init and the methods. Should be one or the other.

**Comment by jerpint** [comment]
> It's weird to have topk in both the init and the methods. Should be one or the other.

@hbertrand agreed, but if we leave it out of the config, then it's up to the front-end to define what that value should be, which is also weird?

File: buster/retriever/base.py (hunk 3 above)
Comment (jerpint) [reply buster/retriever/base.py:43]: fixed

File: buster/retriever/base.py (hunk 2 above)
Comment (jerpint) [reply buster/retriever/base.py:44]: fixed

File: buster/retriever/base.py (hunk 1 above)
Comment (jerpint) [reply buster/retriever/base.py]: fixed

**Review by hbertrand — APPROVED** [review]

**Pull request merged**